src/cls/IPM/General/SemanticVersionExpression/Comparator.cls

```diff
@@ -85,12 +85,14 @@ Method Evaluate(pVersion As %IPM.General.SemanticVersion) As %Boolean
 {
 	Set tEquals = (..Operator [ "=")
   If tEquals {
-    For part = "Major", "Minor", "Patch", "Prerelease" {
+    For part = "Major", "Minor", "Patch", "Prerelease", "Build" {
       Set val1 = $Property(pVersion, part)
       Set val2 = $Property($this, part)
-      Quit:((val1 = "")||(val2 = ""))
       Quit:((val1 = "*")||(val2 = "*"))
-      Set tEquals = tEquals && (val1 = val2)
+	  // Enforce exact match if comparator specifies the part (e.g., Prerelease or Build)
+	  If val2 '= "" {
```

Comment:
Why not keep the original quit in which checks for if either val1 or val2 is empty?